### PR TITLE
Feat/redux saga implemented

### PR DESCRIPTION
**Pull request by prabhat-wednesday**
### Ticket Link

---

### Related Links

---

### Description
Create Itune Container and Make API Call from Saga [file]
---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
Screenshot 2022-09-29 at 12 24 51 PM

---

### CODE REVIEW

File: app/components/Action/index.tsx [hunk 1]
```diff
@@ -1,35 +1,14 @@
...
-      const res = await api.any({ method: 'GET', url: `/search?term=${searchItem}` });
-      setData(res);
-    } catch (error) {
-      console.log(error);
-    }
-    setSearchItem('');
-  };
-
-  console.log(data);
+// const api = create({ baseURL: 'https://itunes.apple.com' });
```
Comment (deepak-wednesday) [inline app/components/Action/index.tsx]: remove comment

File: app/containers/App/index.tsx [hunk 2]
```diff
@@ -36,37 +36,40 @@ const CustomLayout = styled(Layout)`
...
+    //                 return <Component {...updatedProps} />;
+    //               }}
+    //             />
+    //           );
+    //         }}
+    //       />
+    //       <GlobalStyle />
+    //     </Layout.Content>
+    //   </CustomLayout>
+    // </ThemeProvider>
```
Comment (deepak-wednesday) [inline app/containers/App/index.tsx]: @prabhat-wednesday please remove whatever is not required & also look into this files that what are different things that are commented but useful.

File: app/containers/ItuensContainer/apiUtils.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      return res;
+    } catch (error) {
+      return error;
+    }
+  };
+
+  return apiCall;
+};
+
+export default useApiSauce;
```
Comment (deepak-wednesday) [inline app/containers/ItuensContainer/apiUtils.ts:18]: shouldn't this file be in the utils folder ? , also fix typo in filename .

File: app/containers/ItuensContainer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import { getSearchTerm } from './reducer';
+import { useDispatch, connect } from 'react-redux';
+import ituneCallSaga from './saga';
+import { injectSaga } from 'redux-injectors';
+import { AnyAction, compose } from '@reduxjs/toolkit';
+import { setQueryParam } from '@app/utils';
+import history from '@app/utils/history';
+import { selectError, selectLoading, selectDataToShow } from './selector';
+import { isEmpty } from 'lodash-es';
+
```
Comment (deepak-wednesday) [inline app/containers/ItuensContainer/index.tsx:13]: Let's follow this import order 
// import from a third-party library
// import from alias
// relative imports

**Review by deepak-wednesday — CHANGES_REQUESTED** [review]
@prabhat-wednesday , please remove all the consoles , comments & let try to rename the functions , reducers names properly which signifies some meaning .

File: app/containers/ItuensContainer/apiUtils.ts (hunk 3 above)
Comment (prabhat-wednesday) [reply app/containers/ItuensContainer/apiUtils.ts:18]: I have moved this file in service folder in feat/created-UI branch

File: app/containers/ItuensContainer/index.tsx (hunk 4 above)
Comment (deepak-wednesday) [reply app/containers/ItuensContainer/index.tsx:13]: Please check this

File: app/containers/ItuensContainer/apiUtils.ts (hunk 3 above)
Comment (deepak-wednesday) [reply app/containers/ItuensContainer/apiUtils.ts:18]: ok

**Pull request merged**